### ISSUE TITLE
Inconsistent expo-sqlite/next

### ISSUE THREAD

**Issue by abduraufsherkulov** [expo/expo#27619]
### Minimal reproducible example
https://github.com/abduraufsherkulov/expo-sqlite
-

### What platform(s) does this occur on?

Android

### Did you reproduce this issue in a development build?

Yes

### Summary

Updated my codebase to use expo-sqlite/next, however it seems API is working inconsistently. Given I do make requests multiple times to SELECT, it is throwing:

```
[Error: Call to function 'NativeDatabase.prepareAsync' has been rejected.
Caused by: java.lang.NullPointerException: java.lang.NullPointerException]
```

Video (please bear with me, it stops working only after making requests multiple times):
https://github.com/expo/expo/assets/17489128/f9dafaaf-5f5e-4c60-a797-1af8e507c777

### Environment

```
  expo-env-info 1.2.0 environment info:
    System:
      OS: macOS 14.4
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 20.9.0 - ~/.nvm/versions/node/v20.9.0/bin/node
      Yarn: 1.22.11 - /usr/local/bin/yarn
      npm: 10.1.0 - ~/.nvm/versions/node/v20.9.0/bin/npm
      Watchman: 2024.01.22.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.14.3 - /opt/homebrew/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.2, iOS 17.2, macOS 14.2, tvOS 17.2, visionOS 1.0, watchOS 10.2
      Android SDK:
        API Levels: 23, 28, 29, 30, 31
        Build Tools: 28.0.3, 29.0.2, 31.0.0
        System Images: android-31 | Google APIs ARM 64 v8a
    IDEs:
      Android Studio: 2020.3 AI-203.7717.56.2031.7784292
      Xcode: 15.2/15C500b - /usr/bin/xcodebuild
    npmPackages:
      expo: ~50.0.11 => 50.0.11 
      expo-router: ~3.4.8 => 3.4.8 
      react: 18.2.0 => 18.2.0 
      react-native: 0.73.4 => 0.73.4 
    npmGlobalPackages:
      eas-cli: 7.4.0
      expo-cli: 6.3.10
    Expo Workflow: managed
```

### Expo Doctor Diagnostics

-

**Comment by curtcommander** [comment]
I'm seeing this same behavior with expo-sqlite/next on Android

**Comment by brentvatne** [comment]
@abduraufsherkulov - the link to the github repo you shared doesn't exist
<img width="1401" alt="image" src="https://github.com/expo/expo/assets/90494/0843f7e6-1272-4214-a348-bcebab20c7f5">

**Comment by abduraufsherkulov** [comment]
@brentvatne argh, silly me. I made it public now. Could we please reopen now?

**Comment by abduraufsherkulov** [comment]
@brentvatne The reason I added bunch of screens because it is not easily reproducible by just making a call. I also uploaded a video, it should be helpful. Let me know if you need anything else.

**Comment by dylanfcsr** [comment]
This problem often happens to me as well.

**Comment by abduraufsherkulov** [comment]
@dylanfcsr That could be true but I tried with a minimal demo and worked fine. Created two buttons to make requests and smashed them multiple times with 0.01 sec delay. It worked fine but it is somehow failing when I am incorporating expo-router.

**Comment by abduraufsherkulov** [comment]
@brentvatne Hi Brent, could we please re-open it?

**Comment by Kudo** [comment]
the issue should be fixed by `expo-sqlite@13.4.0`. please let me know whether it works for you.

**Comment by abduraufsherkulov** [comment]
@NazarDevsMedia what version are you using?

**Comment by NazarDevsMedia** [comment]
> @NazarDevsMedia what version are you using?
seems it was realesed with
```
expo 50.0.13
sqlite 13.3.0
```
not with 13.4.0 :/

**Comment by claudiofrs** [comment]
same issue. no problem in emulator but when previewing the development mode with expo go, this error exist

**Comment by NazarDevsMedia** [comment]
same with expo: 50.0.17 & expo-sqlite: 13.4.0.

@Kudo if the issue is accepted could we please reopen this?

**Comment by Kudo** [comment]
i resolved the issue by my understanding from the https://github.com/abduraufsherkulov/expo-sqlite repro. if the fix doesn't work with you, i guess it might be a different root cause. please try to recreate a new issue with a repro. that would be super helpful. thanks!

**Comment by zaihdo** [comment]
Experiencing the same issue with expo go on iOS:

 [Error: FunctionCallException: Calling the 'prepareSync' function has failed (at ExpoModulesCore/SyncFunctionDefinition.swift:87)
→ Caused by: SQLiteErrorException: Error code 1: no such table: salah_times (at ExpoSQLite/SQLiteModuleNext.swift:269)

Has there been any resolution yet

**Comment by xiaojin9712** [comment]
I've encountered this issue also, but after review my code, I found that it seems I'm using async method to create a new table, after I changed it to sync method, make sure that task already existed before access it, works for me now.
